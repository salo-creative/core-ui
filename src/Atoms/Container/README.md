# Container

Container is a pure styled component render. Renders a standard `<div>` with themed styling

-------

## Usage

Install

```javascript
yarn add @salo/core-ui
```

Include

```javascript
import Container from '@salo/core-ui/Atoms/Container';
```

Implement as follows

```javascript
<Container>
  // My content
</Container>
```

### For full prop types and usage see storybook info/knobs