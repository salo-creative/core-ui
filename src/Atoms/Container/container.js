import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledContainer = styled.div`
  width: 100%;
  margin: 0 auto;
  position: relative;
  max-width: ${ props => props.width };
  padding: ${ props => props.padding };
`;

const Container = ({ children, ...props }) => {
  return <StyledContainer { ...props }>{ children }</StyledContainer>;
};

Container.defaultProps = {
  padding: '10px',
  width: '1280px'
};

Container.propTypes = {
  padding: PropTypes.string,
  width: PropTypes.string
};

export default Container;