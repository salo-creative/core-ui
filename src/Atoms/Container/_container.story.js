import React from 'react';
import { storiesOf } from '@storybook/react';

// load tests
import { withTests } from '@storybook/addon-jest';
import { withKnobs, text } from '@storybook/addon-knobs';
import results from '../../../.storybook/jest-test-results.json';

// FEATURED COMPONENT
import { Container } from '../../index';

// README //
import README from './README.md';

// Start of story logic
const stories = storiesOf('Atoms | Grid/Container', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests({ results }));
stories.addParameters({ jest: ['container'] });

stories.add(
  'Basic',
  (() => {
    const width = text('Width', '1280px');
    const padding = text('Padding', '0 0 0 0');
    return (
      <Container
        width={ width }
        padding={ padding }
        style={ { background: '#eaeaea' } }
      >
        A lovely Container
      </Container>
    );
  }), { notes: README }
);