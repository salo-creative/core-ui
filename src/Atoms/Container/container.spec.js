import React from 'react';
import 'jest-styled-components';

// HELPERS
import { renderWithTheme } from '../../../test';

// COMPONENT
import Container from './index';

test('Render default Container and styles', async () => {
  // Render
  const { getByTestId, container } = renderWithTheme(
    <Container data-testid='test-container'>A tasty treat</Container>
  );
  const el = getByTestId('test-container');
  // Assert
  expect(el).toHaveTextContent('A tasty treat');
  expect(el).toHaveStyleRule('padding', '10px');
  expect(el).toHaveStyleRule('max-width', '1280px');
  expect(container.firstChild).toMatchSnapshot();
});

test('Render Container and styles with core props', async () => {
  // Render
  const { getByTestId, container } = renderWithTheme(
    <Container
      padding='20px'
      width='200px'
      data-testid='test-container'
    >
      A tasty treat
    </Container>
  );
  const el = getByTestId('test-container');
  // Assert
  expect(el).toHaveStyleRule('padding', '20px');
  expect(el).toHaveStyleRule('max-width', '200px');
  expect(container.firstChild).toMatchSnapshot();
});