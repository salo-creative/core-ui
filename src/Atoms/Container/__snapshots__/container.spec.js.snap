// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Container and styles with core props 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
  position: relative;
  max-width: 200px;
  padding: 20px;
}

<div
  class="c0"
  data-testid="test-container"
  width="200px"
>
  A tasty treat
</div>
`;

exports[`Render default Container and styles 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
  position: relative;
  max-width: 128rem;
  padding: 1rem;
}

<div
  class="c0"
  data-testid="test-container"
  width="128rem"
>
  A tasty treat
</div>
`;
